lib/em/protocols/linetext2.rb

```diff
@@ -57,6 +57,7 @@ def receive_data data
         while remaining_data.length > 0
           if @lt2_mode == :lines
             if ix = remaining_data.index( @lt2_delimiter )
+              delimiter_length = remaining_data.slice(@lt2_delimiter).length
```

Comment:
We could use slice to extract the matching string and then do to a string search to extract the index.  That will add a line or two of code but only execute the regex match once.  It will probably be more efficient in execution.

`if ix_str = remaining_data.slice( @lt2_delimiter )
   ix = remaining_data.index(ix_str)
   delimiter_length = ix_str.length`

Something like that maybe?